wayland-backend/src/sys/client_impl/mod.rs

```diff
@@ -764,7 +764,10 @@ impl InnerBackend {
         data: Arc<dyn ObjectData>,
     ) -> ObjectId {
         let mut guard = self.lock_state();
-        unsafe { self.manage_object_internal(interface, proxy, data, &mut guard) }
+        unsafe {
+            ffi_dispatch!(wayland_client_handle(), wl_proxy_set_queue, proxy, guard.evq);
```

Comment:
One thing to keep in mind is that *in theory* this is racy: changing the event queue of an existing proxy is not thread-safe (which is why the whole "proxy wrapper" stuff exists when creating proxies: so that they are lerictly created on the correct queue).

I think in most cases this is fine, but I think this should still be documented in the public API docs.

The core of the risk, is that if another thread is concurrently reading from the socket while this is being invoked, and the object that is in the process of being managed already received an event, that event may end up enqueued on the old queue, which may in turn cause dispatching issues or unsafety.

This is really a corner-case situation, and mostly of relevance for objects that are susceptible to receive events as soon as they are created, but I think this would be better written down so as to not surprise anyone.

Reply:
Added a safety warning, is this good enough?